src/processors/default/createERC721NFTsFromTransfersProcessor.ts

```diff
@@ -31,43 +32,68 @@ const processorFunction = (contracts: NftFactory[]) =>
         throw 'buildNFTId not specified'
       }
 
+      const contractAddress = formatAddress(contract.id);
+      const fromAddress = formatAddress(item.args!.from);
+      const toAddress = formatAddress(item.args!.to);
       const tokenId = BigInt((item.args!.tokenId as BigNumber).toString());
-      const newMint = item.args!.from === ETHEREUM_NULL_ADDRESS;
+      const nftId = contractType.buildNFTId(contractAddress, tokenId);
+
       transfers.push({
-        id: `${CHAIN}/${item.blockNumber}/${item.logIndex}`,
-        contractAddress: formatAddress(contract.id),
-        from: item.args!.from,
-        to: item.args!.to,
+        id: ethereumTransferId(item.blockNumber, item.logIndex),
+        contractAddress: contractAddress,
+        from: fromAddress,
+        to: toAddress,
         tokenId,
         createdAtEthereumBlockNumber: '' + item.blockNumber,
-        nftId: contractType.buildNFTId(contract.id, tokenId), 
+        nftId: nftId,
         transactionHash: item.transactionHash
       });
-      if (!newMint) {
-        updates.push({
-          id: contractType.buildNFTId(contract.id, tokenId),
-          owner: item.args!.to
+
+      if (!newMint(fromAddress)) {
+        updatedNFTs.push({
+          id: nftId,
+          owner: toAddress
         })
+        updatedNftsCollectors.push({ nftId: nftId, collectorId: fromAddress, amount: -1 })
+        updatedNftsCollectors.push({ nftId: nftId, collectorId: toAddress, amount: 1 })
         return undefined;
       }
       newNFTs.push({
-        id: contractType.buildNFTId(contract.id, tokenId),
+        id: nftId,
         createdAtEthereumBlockNumber: '' + item.blockNumber,
-        contractAddress: formatAddress(contract.id),
+        contractAddress: contractAddress,
         tokenId,
         platformId: contract.platformId,
-        owner: item.args!.to,
+        owner: toAddress,
         approved: contract.autoApprove
       });
+      updatedNftsCollectors.push({ nftId: nftId, collectorId: toAddress, amount: 1 })
     });
 
+    // update Collectors table
```

Comment:
The overall strategy is to effectively:
1. collect all transfers and assign to collectors  (+1 amount for receiver, -1 amount for sender)
2. add any mints (+1 for receiver)
3. update DB with any missing collector addresses
4. update/add any NftsCollectors with the appropriate amounts

Very brute force but I think it can work with a bit of cleanup and optimization

Reply:
i think it looks alright - i see the stronger motivation for separating this out into a different processor now as u suggested, since that would simplify it further but the current implementation still looks good so i think we can keep it.

suggestions:
 - move line 75 down, ie, the Table.collectors insert so that we do all the inserts together instead of interweaving logic/getRecords and inserts
 - could you move both the Table.collectors insert and Table.nftsCollectors inserts down below the newNFTs insert to solve your ordering problem related to the foreignkey on the nftCollectors table?

Reply:
I've colocated the inserts. Also tried moved those two lower down in the order but oddly enough it doesn't fix the problem 😕 

I think there may be some conditions where the above NFTs are filtered out before the insert/upsert. All I can see is  line 89
```ts
await clients.db.insert(Table.nfts, newNFTs.filter(n => !!n), { ignoreConflict: 'id' });
```
But that's just clearing empty ones, no? Am I missing something elsewhere?

Reply:
yeah that is weird. maybe something is wrong/may need to debug which things are failing and why